test/alternator/test_returnvalues.py

```diff
@@ -83,6 +83,47 @@ def test_put_item_returnvalues(test_table_s):
     with pytest.raises(ClientError, match='ValidationException'):
         test_table_s.put_item(Item={'p': p, 'a': 'hello'}, ReturnValues='none')
 
+def skip_if_returnvalues_on_condition_check_not_supported():
+    import botocore
+    from packaging.version import Version
+    # This release added support for ReturnValuesOnConditionCheckFailure
+    if (Version(botocore.__version__) < Version('1.29.164')):
+        pytest.skip("Botocore version 1.29.164 or above required to run this test")
+
+# Testing ReturnValuesOnConditionCheckFailure feature which returns values only
+# on failed condition expression.
+def test_put_item_returnvalues_on_condition_check_failure(test_table_s):
+    skip_if_returnvalues_on_condition_check_not_supported()
+    p = random_string()
+    test_table_s.put_item(Item={'p': p, 's': 'dog'})
+    # Succesful conditional put doesn't return values.
+    ret=test_table_s.put_item(Item={'p': p, 's': 'cat'},
+        ReturnValuesOnConditionCheckFailure='ALL_OLD',
+        ConditionExpression='s = :v1',
+        ExpressionAttributeValues={':v1' : 'dog'})
+    assert not 'Attributes' in ret # Ordinary field not present.
+    assert not 'Item' in ret # Field used for error responses also not present.
+    # Failed conditional put returns old values.
+    with pytest.raises(test_table_s.meta.client.exceptions.ConditionalCheckFailedException) as err:
```

Comment:
looks like this is the only non string way (https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html), because aws error classes are dynamically generated, I am assuming they call some special endpoint to construct them or make them on the go as they receive error responses from AWS.